pkg/controller/operators/operator_controller.go

```diff
@@ -152,9 +153,11 @@ func (r *OperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			return ctrl.Result{Requeue: true}, nil
 		}
 	} else {
-		if err := r.Status().Update(ctx, operator.Operator); err != nil {
-			log.Error(err, "Could not update Operator status")
-			return ctrl.Result{Requeue: true}, nil
+		if !reflect.DeepEqual(in.Status, operator.Operator.Status) {
```

Comment:
Nit: it's probably better to use `equality.Semantic.DeepEqual` from `k8s.io/apimachinery/pkg/api/equality` since that correctly handles non-material differences in the comparison.

https://pkg.go.dev/k8s.io/apimachinery/pkg/api/equality#pkg-variables

Reply:
cc @asmacdo this would have been good to include in your operator's best practices doc!

Reply:
I'll update that doc, TIL